facebook/Docusaurus
# Fix assets not found in Express server

## Motivation

Fixes #600

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
1. Create an image file in `docs/assets/docusaurus.png`

2. Paste this to `docs/getting-started-installation.md` content
```![docusaurus](/docs/assets/docusaurus.png)```

3. Run server on windows
```
cd website
npm start
```
http://localhost:3000/docs/en/next/installation.html

Before this PR
<img width="917" alt="before" src="https://user-images.githubusercontent.com/17883920/40869774-2e651ef0-6653-11e8-9a44-9ae4c2c96d67.PNG">

After this PR
<img width="960" alt="after" src="https://user-images.githubusercontent.com/17883920/40869786-67261578-6653-11e8-8b14-85b165d6093a.PNG">

Explanation:

According to https://expressjs.com/en/starter/static-files.html 
The function signature is:
```js
express.static(root, [options])
```

Previous code
```js
  app.use(
    join(siteConfig.baseUrl, 'docs', 'assets'),
    express.static(join(CWD, '..', readMetadata.getDocsPath(), 'assets'))
  );
  app.use(
    join(siteConfig.baseUrl, 'blog', 'assets'),
    express.static(join(CWD, 'blog', 'assets'))
  );
```

We should not use path.join on Express root argument. Express.js expect something like `/docs/assets/` as a root argument. `path.join` in Unix systems work differently when in Windows work.

Example:
```
console.log(path.join(siteConfig.baseUrl, 'docs', 'assets'));
```

In UNIX `/docs/assets`
In Windows `\docs\assets`
